practicas/Aleksei115/P1/1_niveles_linguisticos.py

```diff
@@ -0,0 +1,554 @@
+    # "zh_hans": "Mandarin (Simplified)",  # Usa caracteres chinos simplificados
+    # "zh_hant": "Mandarin (Traditional)"  # Usa caracteres chinos tradicionales
+}
+
+iso_lang_codes = list(lang_codes.keys())
+
+
+# In[7]:
+
+
+def get_formated_string(code: str, name: str):
+    return f"[b]{name}[/b]\n[yellow]{code}"
+
+def parse_response(response: str) -> dict:
+    """Parse text response from ipa-dict to python dict
+
+    Each row have the format:
+    [WORD][TAB]/[IPA]/(, /[IPA]/)?
+
+    Parameters
+    ----------
+    response: str
+        ipa-dict raw text
+
+    Returns
+    -------
+    dict:
+        A dictionary with the word as key and the phonetic
+        representations as value
+    """
+    ipa_list = response.rstrip().split("\n")
+    result = {}
+    for item in ipa_list:
+        if item == '':
+            continue
+        item_list = item.split("\t")
+        result[item_list[0]] = item_list[1]
+    return result
+
+
+def download_ipa_corpus(iso_lang: str) -> str:
+    """Get ipa-dict file from Github
+
+    Parameters:
+    -----------
+    iso_lang:
+        Language as iso code
+
+    Results:
+    --------
+    dict:
+        Dictionary with words as keys and phonetic representation
+        as values for a given lang code
+    """
+    print(f"Downloading {iso_lang}", end="::")
+    response = r.get(IPA_URL.format(lang=iso_lang))
+    status_code = response.status_code
+    print(f"status={status_code}")
+    if status_code != http.HTTPStatus.OK:
+        print(f"ERROR on {iso_lang} :(")
+        return ""
+    return response.text
+
+
+def get_corpora() -> dict:
+    """Download corpora from ipa-dict github
+
+    Given a list of iso lang codes download available datasets.
+
+    Returns
+    -------
+    dict
+        Lang codes as keys and dictionary with words-transcriptions
+        as values
+    """
+    return {code: parse_response(download_ipa_corpus(code)) for code in iso_lang_codes}
+
+
+def get_ipa_transcriptions(word: str, dataset: dict) -> list[str]:
+    """Search for a word in an IPA phonetics dict
+
+    Given a word this function return the IPA transcriptions
+
+    Parameters:
+    -----------
+    word: str
+        A word to search in the dataset
+    dataset: dict
+        A dataset for a given language code
+
+    Returns
+    -------
+    list[str]:
+        List with posible transcriptions if any,
+        else an empty list
+    """
+    return dataset.get(word.lower(), "").split(", ")
+
+data = get_corpora()
+
+
+# In[19]:
+
+
+data = get_corpora()
+
+
+# In[41]:
+
+
+N_SIMILAR = 3
+def aproximacion_fonetica(funcion_simitud):
+  '''
+  Función que regresa aproximaciones de transcripciones fonéticas para palabras
+  que no están en el lexicón
+
+  Input:
+    funcion_simitud: funcion a usar para encontrar las palabras más
+    similares.
+
+  Output:
+    Regresa la transcripción fonética, ya sea la que le pertence o una
+    aproximación.
+  '''
```

Comment:
Hay convenciones en cuanto a los docstrings te recomiendo seguir alguno. El que uso yo es el de `numpy`:    

- https://stackoverflow.com/questions/3898572/what-are-the-most-common-python-docstring-formats